lib/index.ts

```diff
@@ -0,0 +1,70 @@
+import { Server, Plugin } from "@hapi/hapi";
+import * as HapiAuthJwt from "hapi-auth-jwt2";
+import verifyJWT, { Validator } from "./verifyJWT";
+import * as keystoresFromFile from "./keystores.json";
+import { assert } from "@hapi/hoek";
+import makeIssueTokenRoute from "./make-issue-token-route";
+
+export type ClientSecrets = {
+  [key: string]: string;
+};
+
+type HapiOidcOptions = {
+  tokenEndpoint?: string;
+  clients?: ClientSecrets;
+  fetchKeystore?: () => object | Promise<object>;
```

Comment:
I'm not bothering with typing the keystore object. It's a big ol' thing and there aren't public types for it that I could find quickly

Reply:
You can have the `resolveJsonModule` compiler option set to true, so you can do this;
```
import keystoresFromFile from "./keystores.json";
type Keystore = typeof keystoresFromFile;
```
![image](https://user-images.githubusercontent.com/8016126/87461477-614dba80-c5c3-11ea-8970-cdeb262981e7.png)

Reply:
Is that a good idea considering that the shapes are different depending on the types of keys in the keystore? That type definition is incomplete because there are a bunch of different algorithms and key types that would have a bunch of different values.

Reply:
I believe you are right, you don't want to do what I said. I'm not that familiar with how keystores work.

Maybe something like this would be true of a keystore shape?
```
type Keystore = { keys: { [key: string]: string }[] }
```

That is saying that a keystore is an object with a property called "keys" that consists of an array of objects in which keys (properties) and values are strings.

Reply:
Yeah good idea, i can do that. Also `kty` and `kid` are for sure always present